lib/nuget-parser/cli/dotnet.ts

```diff
@@ -109,8 +116,9 @@ export async function publish(
   const tempDir = fs.mkdtempSync(
     path.join(os.tmpdir(), `snyk-nuget-plugin-publish-csharp-`),
   );
-  args.push('--output');
-  args.push(tempDir);
+  // See https://learn.microsoft.com/en-us/dotnet/core/compatibility/sdk/7.0/solution-level-output-no-longer-valid#recommended-action
+  // about why we're not using `--output` for this.
+  args.push(`--property:PublishDir=${tempDir}`);
```

Comment:
We were seeing 

> warning NETSDK1194: The "--output" option isn't supported when building a solution. Specifying a solution-level output path results in all projects copying outputs to the same directory, which can lead to inconsistent builds.

And 

> The process cannot access the file '<path>' because it is being used by another process.

In some customer workflows. Suspecting this to be the cause, although cannot verify, yet. But end result should be the same.